wallet/src/de/schildbach/wallet/ui/WalletActivity.java

```diff
@@ -529,7 +531,31 @@ public void handleRestoreWallet() {
     }
 
     public void handleBackupWalletToSeed() {
-        BackupWalletToSeedDialogFragment.show(getSupportFragmentManager());
+        handleVerifySeed();
+    }
+
+    private void handleVerifySeed() {
+        final int AUTH_REQUEST_CODE_VIEW_RECOVERYPHRASE = 1;
+        DecryptSeedSharedModel decryptSeedSharedModel = ViewModelProviders.of(this).get(DecryptSeedSharedModel.class);
+        decryptSeedSharedModel.getOnDecryptSeedCallback().observe(this, new Observer<Pair<Integer, DeterministicSeed>>() {
+
+            @Override
+            public void onChanged(Pair<Integer, DeterministicSeed> data) {
+                switch (data.getFirst()) {
+                    case AUTH_REQUEST_CODE_VIEW_RECOVERYPHRASE:
+                        startVerifySeedActivity(data.getSecond());
+                        break;
+                }
+            }
+        });
+        DecryptSeedWithPinDialog.show(this, AUTH_REQUEST_CODE_VIEW_RECOVERYPHRASE);
+    }
+
+    private void startVerifySeedActivity(DeterministicSeed seed) {
+        List<String> mnemonicCode = seed.getMnemonicCode();
+        String [] seedArray = mnemonicCode.toArray(new String[0]);
+        Intent intent = VerifySeedActivity.Companion.createIntent(this, seedArray);
```

Comment:
You can add the `@JvmStatic` annotation to `createIntent(context: Context, seed: Array<String>)` method definition in `VerifySeedActivity` class in order to be able to simply call `VerifySeedActivity.createIntent(this, seedArray)` without `Companion`.

Reply:
Thank you for this idea.

Reply:
https://github.com/dashevo/dash-wallet/pull/300/commits/82fc9d131051506372c3c9f69ac6bb890066e896